### DISCUSSION TITLE
Filebeat and Winlogbeat

### DISCUSSION THREAD

**Discussion by Mav1814** [Security-Onion-Solutions/securityonion#9054] [Unsupported Versions] [answered]
Hello 
I am currently performing tests with a distributed Security onion deployment, I have a manager, a forwarder and a search node.

I have configured two machines to send logs via filebeat (linux) and winlogbeat (windows). But after the default configuration (I had already tested in standalone), it shows me in both services logs.

On filebeat logs
`Nov 02 09:46:30 debian filebeat[12532]: 2022-11-02T09:46:30.224Z        ERROR        [publisher_pipeline_output]        pipeline/output.go:154        Failed to connect to backoff(async(tcp://x.x.x.4:5044)): dial tcp x.x.x.x:5044: connect: no route to host
Nov 02 09:46:30 debian filebeat[12532]: 2022-11-02T09:46:30.225Z        INFO        [publisher_pipeline_output]        pipeline/output.go:145        Attempting to reconnect to backoff(async(tcp://x.x.x.x:5044)) with 9342 reconnect attempt(s)
`
On winlogbeat logs
`A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond`

In the configuration file of both machines is the forwarder's IP, initially I thought the problem was with the firewall, even after adding the addresses of the machines, and doing allow on the manager the result was the same of above. Iptables bellow.

- `ACCEPT     tcp  --  x.x.x.x252          0.0.0.0/0            tcp dpt:5044
- ACCEPT     udp  --  x.x.x.x.252          0.0.0.0/0            udp dpt:1514
- ACCEPT     tcp  --  x.x.x.x.252          0.0.0.0/0            tcp dpt:1514
`
Wazuh and osquery works on both machines. 

I am used to using filebeat and winlogbeat, I have encountered this problem before with ELK Stack, but I have tried to solve it and can't figure out where the problem is.

I can ping the IPs from the machines to the forwarder and vice-versa. Netstat doesn't show anything on both IPs. 

Anyone with some ideas what might be the problem?

**Comment by Mav1814** [comment]
UPDATE:

if i send directly the machines trough the manager IP it works. But with the forwarder IP it doesn't .  
On the architecture for distributed deployment, it shows that the forwarder sends the logs from filebeat to logstash. And isn't working.

**Comment by dougburks** [comment, marked as answer]
Please see https://docs.securityonion.net/en/2.3/firewall.html#host-firewall.

**Reply by Mav1814** [reply]
Hi @dougburks Thank you for the response. Sorry for  being a bit rude, but sending only the documentation its not helpful at all.. I already see all the documentation of Security Onion.  didn't help.  :/ 
I have put here information about the problem in order to help to find where the problem is... 

On  the firewall i did this for the filebeat and winlogbeat. Before i have done it for the Wazuh agents and worked. 

1. Create the firewall entry for the agent with sudo so-firewall include beats_endpoint IP
2. Edit the minion pillar file for the sensor you want to send the logs to /opt/so/saltstack/local/pillar/minions/*sensor.sls
3. Add the following to the pillar file

```
firewall:
		  assigned_hostgroups:
		    chain:
		      DOCKER-USER:
		        hostgroups:
		          beats_endpoint:
		            portgroups:
		              - portgroups.beats_5044
```

4. Apply the firewall state to the sensor with sudo salt '*sensor*' state.apply firewall
5. Start/Restart the agent on the endpoint and check the logs for connection

And Didn't work also.

**Reply by Mav1814** [reply]
More, on the documentation in the architecture, it says on the forwarder node:

A forward node is a sensor that forwards all logs via [Filebeat](https://docs.securityonion.net/en/2.3/filebeat.html#filebeat) to [Logstash](https://docs.securityonion.net/en/2.3/logstash.html#logstash) on the manager node, where they are stored in [Elasticsearch](https://docs.securityonion.net/en/2.3/elasticsearch.html#elasticsearch) on the manager node or a search node (if the manager node has been configured to use a search node).

The configuration of the filebeat on the agent machine, is simple.

 ```
    filebeat.inputs enable 
    setup.dashboards.enabled setting is commented.
    output.elasticsearch setting is commented.
    Enable the output.logstash and configure it to send logs to port 5044 on your forwarder node.
```
Then restarted the service and shows the errors showed above.  So or i'm doing something wrong, or what is on the documentation showed above is false.

**Reply by dougburks** [reply]
I mis-read your original post. If you are configuring agents to send to Logstash, then you'll need to send to a node which runs Logstash. Forward nodes do not run Logstash, so you would need to send to either a manager node or a receiver node.

For example, from https://docs.securityonion.net/en/2.3/beats.html#winlogbeat:
> Enable the output.logstash output and configure it to send logs to port 5044 on your management node.

**Reply by Mav1814** [reply]
Hi again. 
So let me see if i understand,
  
1. A forwarder node has described on the image in the documentation architecture for distributed deployment is used to receive all the logs of the agents machines right? 
2. It receives all the network sensor components right? 
2.1. Meaning that all the agents machines when configured with filebeat, winlogbeat, osquery, wazuh, they will send logs, etc to the forwarder node and than the forwarder will sent  trough filebeat to logstash that is on the manager right?
3. If i will configure the manager node to do this work, what is the purpose of the forwarder node? 
4. For what i understand the receiver node is used  to continue to be processed by search nodes in the event the manager node is offline.

**Reply by dougburks** [reply]
> 1\. A forwarder node has described on the image in the documentation architecture for distributed deployment is used to receive all the logs of the agents machines right?

Are you referring to https://docs.securityonion.net/en/2.3/_images/distributed.png? If so, this image is designed to show that Forward Nodes generate their own logs using osquery, Strelka, Suricata, Wazuh, and Zeek and then forward those logs to the manager. There are no endpoint agents shown in this diagram.

> 2\. It receives all the network sensor components right?

Forward nodes are primarily designed to sniff traffic from a tap or span port, analyze that traffic using Suricata and Zeek, and then forward those Suricata and Zeek logs to the manager.

> 2\.1. Meaning that all the agents machines when configured with filebeat, winlogbeat, osquery, wazuh, they will send logs, etc to the forwarder node and than the forwarder will sent  trough filebeat to logstash that is on the manager right?

Filebeat and Winlogbeat must connect to Logstash but Logstash does not run on a forward node. Filebeat and Winlogbeat must therefore connect to a manager or receiver node.

> 3\. If i will configure the manager node to do this work, what is the purpose of the forwarder node?

Forward nodes are primarily designed to sniff traffic from a tap or span port, analyze that traffic using Suricata and Zeek, and then forward those Suricata and Zeek logs to the manager.

> 4\. For what i understand the receiver node is used  to continue to be processed by search nodes in the event the manager node is offline.

Yes, receiver nodes and the manager node run Logstash and are therefore capable of accepting logs from Elastic agents.

**Reply by Mav1814** [reply]
WOW.. 
Until now I had got it all wrong. The interpretation I made about the image is that the forwarder had the role of receiving the logs from the agents and then sending them to the manager.

So, i have more questions, i'm sorry and thank you to help. 

1. Suppose that i have a distributed deployment, a manager, a forwarder and a search node initially. I have n+1 agents, i want them to send the logs via filebeat, wazuh, winlogbeat and osquery what is the best implementation?
2. I need only one receiver for multiple agents in order to send it to the manager?? 

I'm studying SO to implement here on my organization, but it has been a challenge :p

**Reply by dougburks** [reply]
> 1\. Suppose that i have a distributed deployment, a manager, a forwarder and a search node initially. I have n+1 agents, i want them to send the logs via filebeat, wazuh, winlogbeat and osquery what is the best implementation?

You can send logs using any of these but you typically don't want to install multiple agents on the same endpoint. Here are some guidelines for choosing which agent is best for you:
https://docs.securityonion.net/en/2.3/host.html

> 2\. I need only one receiver for multiple agents in order to send it to the manager??

You can start with just a manager node. If you need to expand, you can add one or more receiver nodes depending on how many agents and how much log volume you have:
https://docs.securityonion.net/en/2.3/architecture.html#receiver-node

**Reply by Mav1814** [reply]
I will test as u suggested thank you.

Again i appreciate all the help doug.
